Sources/Rendering/Core/Picker/index.js

```diff
@@ -89,19 +89,144 @@ function vtkPicker(publicAPI, model) {
     return Math.sqrt(tolerance);
   }
 
+  /**
+   * Perform picking on the given renderer, given a ray defined in world coordinates.
+   * @param {*} renderer
+   * @param {*} tolerance
+   * @param {*} p1World
+   * @param {*} p2World
+   * @returns true if we picked something else false
+   */
+  function pick3DInternal(renderer, tolerance, p1World, p2World) {
+    const p1Mapper = new Float64Array(4);
+    const p2Mapper = new Float64Array(4);
+
+    const ray = [];
+    const hitPosition = [];
+
+    //  Loop over all props.  Transform ray (defined from position of
+    //  camera to selection point) into coordinates of mapper (not
+    //  transformed to actors coordinates!  Reduces overall computation!!!).
+    //  Note that only vtkProp3D's can be picked by vtkPicker.
+    const props = model.pickFromList ? model.pickList : renderer.getActors();
+
+    props.forEach((prop) => {
+      const mapper = prop.getMapper();
+      const pickable =
+        prop.getNestedPickable() &&
+        prop.getNestedVisibility() &&
+        prop.getProperty().getOpacity() > 0.0;
```

Comment:
are you sure there always a property in a prop ?
In case of image actors, I think `getProperty()` returns `null`, but `getImageProperty()` does not. Isn't it ?

Reply:
There should always be a property.  
`getProperty` is defined in both `vtkActor` and `vtkVolume`, which to my knowledge are the only two concrete implementation of `vtkProp3D`.  
Also, it is specified that the function always return `vtkProperty` (for `vtkActor`) or `vtkVolumeProperty` (for `vtkVolume`)
that being said, it is a problem that `getProperty` is not defined in `vtkProp3D` itself, so in theory other subclasses could be created and not implement `getProperty`.  
Lastly, this line is a problem for the call to `getOpacity`, which does not exists on a `vtkVolumeProperty`.  I'm going to look for a fix.